flink-connectors/flink-connector-cassandra/src/test/java/org/apache/flink/streaming/connectors/cassandra/CassandraConnectorITCase.java

```diff
@@ -206,6 +209,79 @@ protected Cluster buildCluster(Cluster.Builder builder) {
         }
     }
 
+    private static String setTableNameInPojo(Class<?> claz, String testName) throws Exception {
+        Table clazTableAnnotation = claz.getAnnotation(Table.class);
+        final String newName = clazTableAnnotation.name() + testName;
+        // BEWARE it is JDK 8 specific
+        final Method annotationDataMethod = Class.class.getDeclaredMethod("annotationData", null);
```

Comment:
Have you looked into using libraries to achieve this? Like byte-buddy, asm etc.

An UNTESTED prototype:
```
    @Test
    public void testCassandraPojoAtLeastOnceSink() throws Exception {
        final String someName = "test";
        session.execute(CREATE_TABLE_QUERY.replace(TABLE_NAME_VARIABLE, someName));

        final Class<? extends Pojo> load =
                new ByteBuddy()
                        .subclass(Pojo.class)
                        .annotateType(createTableAnnotation(KEYSPACE, someName))
                        .make()
                        .load(getClass().getClassLoader())
                        .getLoaded();

        writeInstances(load);

        ResultSet rs = session.execute(SELECT_DATA_QUERY.replace(TABLE_NAME_VARIABLE, someName));
        Assert.assertEquals(20, rs.all().size());
    }

    private Table createTableAnnotation(@Nullable String keySpace, String tableName) {
        return new Table() {
            @Override
            public Class<? extends Annotation> annotationType() {
                return Table.class;
            }

            @Override
            public String keyspace() {
                return keySpace;
            }

            @Override
            public String name() {
                return tableName;
            }

            @Override
            public boolean caseSensitiveKeyspace() {
                return false;
            }

            @Override
            public boolean caseSensitiveTable() {
                return false;
            }

            @Override
            public String writeConsistency() {
                return "";
            }

            @Override
            public String readConsistency() {
                return "";
            }
        };
    }

    private <T> void writeInstances(Class<T> clazz) throws Exception {
        final Constructor<T> constructor =
                clazz.getDeclaredConstructor(String.class, Integer.class, Integer.class);
        CassandraPojoSink<T> sink = new CassandraPojoSink<>(clazz, builderForWriting);
        try {
            sink.open(new Configuration());
            for (int x = 0; x < 20; x++) {
                sink.send(constructor.newInstance(UUID.randomUUID().toString(), x, 0));
            }
        } finally {
            sink.close();
        }
    }
```

Reply:
Actually my comment about java version in the code is not accurate, I should have written `JDK8+ COMPLIANT` instead of `BEWARE it is JDK 8 specific`. Indeed I tested it in JDK 11 and it works as well. I will just remove the comment as the whole code is JDK8+. But anyway, using ByteBuddy seems less error prone, I'll use it instead, thanks for the snippet.

Reply:
@zentol I did the replacement of the native reflection code to ByteBuddy. But it seems that ByteBuddy needs to subclass the Pojo class to be able to manipulate its annotations. The problem with that is that the Cassandra mapper downstream no longer sees the fields defined in the super-class (Pojo) if we use ByteBuddy annotated sub-class. WDYT ?

Reply:
I'll tinker a bit!

Reply:
Thanks Chesnay for the patch !
Sorry, you based your work on my partially updated to ByteBuddy code. In the meantime I pushed a fully migrated code. 
The big difference I see in your code is using ByteBuddy `redefine()` instead of `subClass()` that is precisely what I was looking for, thanks ! I'll update my latest code with it.

Reply:
@zentol all done, PTAL